src/exporter/DaemonMetricCollector.cc

```diff
@@ -157,6 +156,52 @@ void DaemonMetricCollector::dump_asok_metrics() {
         dump_asok_metric(perf_info, perf_values, name, labels);
       }
     }
+    // fetch labeled perf counters if config is set to true
+    bool labeledperf = g_conf().get_val<bool>("exporter_get_labeled_counters");
+    if (labeledperf) {
+       std::string counter_dump_response =
+        asok_request(sock_client, "counter dump", daemon_name);
+      if (counter_dump_response.size() == 0) {
+          failures++;
+          continue;
+      }
+      std::string counter_schema_response =
+          asok_request(sock_client, "counter schema", daemon_name);
+      if (counter_schema_response.size() == 0) {
+        failures++;
+        continue;
+      }
+
+      json_object counter_dump = boost::json::parse(counter_dump_response).as_object();
+      json_object counter_schema = boost::json::parse(counter_schema_response).as_object();
+
+      for (auto &labeled_perf : counter_schema) {
+        std::string labeled_perf_group = {labeled_perf.key().begin(), labeled_perf.key().end()};
+        json_object labeled_perf_group_object = labeled_perf.value().as_object();
+        auto counters = labeled_perf_group_object["counters"].as_object();
+        auto counters_labels = labeled_perf_group_object["labels"].as_object();
+        auto labeled_perf_group_counters = counter_dump[labeled_perf_group].as_object()["counters"].as_object();
+        labels_t labels;
+
+        for(auto &label: counters_labels) {
+          std::string label_key = {label.key().begin(), label.key().end()};
+          labels[label_key] = quote(label.value().as_string().c_str());
+        }
+        labels["ceph_daemon"] = quote(daemon_name);
+        for (auto &counter : counters) {
+          json_object counter_group = counter.value().as_object();
+          if (counter_group["priority"].as_int64() < prio_limit) {
+            continue;        
+          }
+          std::string counter_name_init =  {counter.key().begin(), counter.key().end()};
+          std::string counter_name = "ceph_" + labeled_perf_group + "_" + counter_name_init;
+          std::replace_if(counter_name.begin(), counter_name.end(), is_hyphen, '_');
+
+          auto perf_values = labeled_perf_group_counters.at(counter_name_init);
+          dump_asok_metric(counter_group, perf_values, counter_name, labels);
```

Comment:
I think this means that unlabeled perf counters are dumped twice because `counter dump` output includes both labeled and unlabeled perf counters -- precisely so that consumers like ceph-exporter don't need to scrape the output of two different commands, getting potentially inconsistent results.

Reply:
> I think this means that unlabeled perf counters are dumped twice because `counter dump` output includes both labeled and unlabeled perf counters -- precisely so that consumers like ceph-exporter don't need to scrape the output of two different commands, getting potentially inconsistent results.

Yes you're right, thanks for pointing it out! I completely forgot that it has both labeled and unlabeled counters in command output. I've addressed this in the PR https://github.com/ceph/ceph/pull/50718.